### PR TITLE
resort to one-element mutable arrays as a workaround for aliased `var` field

### PR DESCRIPTION
**Pull request by ggreif**
This is a response to the review comment about better not use experimental features in production code.

Personally I am not sure this is an issue :-)

### CODE REVIEW

**Review by crusso — APPROVED** [review]
Why not use an object? But I'll let it go.... thanks for making change.

File: src/prelude/internals.mo [hunk 1]
```diff
@@ -481,26 +486,26 @@ func @timer_helper() : async () {
...
+            if (n.expire[0] > 0 and n.expire[0] <= now and gathered < thunks.size()) {
                 thunks[gathered] := ?(n.job);
                 switch (n.delay) {
                     case (?delay) {
                         // re-add the node
-                        let expire = n.expire + delay;
+                        let expire = n.expire[0] + delay;
                         // N.B. insert only works on pruned nodes
                         func reinsert(m : ?@Node) : @Node = switch m {
-                            case null ({ n with var expire; ante = null; dopo = null });
```
Comment (kentosugama) [inline src/prelude/internals.mo:492]: So is this the operation that was problematic? What is particular about aliased `var` fields that is a problem?

**Comment by ggreif** [comment]
> Why not use an object?

Objects would be even bigger in the heap (array: 12 bytes, object: 16 bytes and static heap vocabulary) and add even more syntax. Contrast this with mutable cells: 8 bytes.

**Comment by nomeata** [comment]
And objects are more expensive to index into.

**Pull request merged**